File: Bridge/System/IO/Path.cs
using Bridge;

namespace System.IO
{

    public static class Path
    {

        public static readonly char[] InvalidPathChars;
        public static readonly char AltDirectorySeparatorChar;
        public static readonly char DirectorySeparatorChar;
        public static readonly char PathSeparator;
        internal static readonly string DirectorySeparatorStr;
        public static readonly char VolumeSeparatorChar;

        internal static readonly char[] PathSeparatorChars;
        
        public static string ChangeExtension(string path, string extension)
        {
            throw new NotImplementedException();
        }

        public static string Combine(string path1, string path2)
        {
            throw new NotImplementedException();
        }

        internal static string CleanPath(string s)
        {
            throw new NotImplementedException();
        }

        public static string GetDirectoryName(string path)
        {
            throw new NotImplementedException();
        }

        public static string GetExtension(string path)
        {
            throw new NotImplementedException();
        }

        public static string GetFileName(string path)
        {
            throw new NotImplementedException();
        }

        public static string GetFileNameWithoutExtension(string path)
        {
            throw new NotImplementedException();
        }

        public static string GetFullPath(string path)
        {
            throw new NotImplementedException();
        }

        internal static String GetFullPathInternal(String path)
        {
            throw new NotImplementedException();
        }
        
        internal static string GetFullPathName(string path)
        {
            throw new NotImplementedException();
        }

        internal static string WindowsDriveAdjustment(string path)
        {
            throw new NotImplementedException();
        }

        internal static string InsecureGetFullPath(string path)
        {
            throw new NotImplementedException();
        }

        internal static bool IsDirectorySeparator(char c)
        {
            throw new NotImplementedException();
        }

        public static string GetPathRoot(string path)
        {
            throw new NotImplementedException();
        }

        public static string GetTempFileName()
        {
            throw new NotImplementedException();
        }

        public static string GetTempPath()
        {
            throw new NotImplementedException();
        }
        
        public static bool HasExtension(string path)
        {
            throw new NotImplementedException();
        }

        public static bool IsPathRooted(string path)
        {
            throw new NotImplementedException();
        }

        public static char[] GetInvalidFileNameChars()
        {
            throw new NotImplementedException();
        }

        public static char[] GetInvalidPathChars()
        {
            throw new NotImplementedException();
        }

        public static string GetRandomFileName()
        {
            throw new NotImplementedException();
        }
        
        public static string Combine(params string[] paths)
        {
            throw new NotImplementedException();
        }

        public static string Combine(string path1, string path2, string path3)
        {
            throw new NotImplementedException();
        }

        public static string Combine(string path1, string path2, string path3, string path4)
        {
            throw new NotImplementedException();
        }

        internal static void Validate(string path)
        {
            throw new NotImplementedException();
        }

        internal static void Validate(string path, string parameterName)
        {
            throw new NotImplementedException();
        }

        internal static string DirectorySeparatorCharAsString
        {
            get
            {
                throw new NotImplementedException();
            }
        }
    }
}

File: Bridge/Resources/.generated/system/iO/path.js
    Bridge.define("System.IO.Path", {
        statics: {
            fields: {
                invalidPathChars: null,
                altDirectorySeparatorChar: 0,
                directorySeparatorChar: 0,
                pathSeparator: 0,
                directorySeparatorStr: null,
                volumeSeparatorChar: 0,
                pathSeparatorChars: null
            },
            props: {
                DirectorySeparatorCharAsString: {
                    get: function () {
                        throw new System.NotImplementedException();
                    }
                }
            },
            methods: {
                changeExtension: function (path, extension) {
                    throw new System.NotImplementedException();
                },
                combine: function (path1, path2) {
                    throw new System.NotImplementedException();
                },
                combine$3: function (paths) {
                    if (paths === void 0) { paths = []; }
                    throw new System.NotImplementedException();
                },
                combine$1: function (path1, path2, path3) {
                    throw new System.NotImplementedException();
                },
                combine$2: function (path1, path2, path3, path4) {
                    throw new System.NotImplementedException();
                },
                cleanPath: function (s) {
                    throw new System.NotImplementedException();
                },
                getDirectoryName: function (path) {
                    throw new System.NotImplementedException();
                },
                getExtension: function (path) {
                    throw new System.NotImplementedException();
                },
                getFileName: function (path) {
                    throw new System.NotImplementedException();
                },
                getFileNameWithoutExtension: function (path) {
                    throw new System.NotImplementedException();
                },
                getFullPath: function (path) {
                    throw new System.NotImplementedException();
                },
                getFullPathInternal: function (path) {
                    throw new System.NotImplementedException();
                },
                getFullPathName: function (path) {
                    throw new System.NotImplementedException();
                },
                windowsDriveAdjustment: function (path) {
                    throw new System.NotImplementedException();
                },
                insecureGetFullPath: function (path) {
                    throw new System.NotImplementedException();
                },
                isDirectorySeparator: function (c) {
                    throw new System.NotImplementedException();
                },
                getPathRoot: function (path) {
                    throw new System.NotImplementedException();
                },
                getTempFileName: function () {
                    throw new System.NotImplementedException();
                },
                getTempPath: function () {
                    throw new System.NotImplementedException();
                },
                hasExtension: function (path) {
                    throw new System.NotImplementedException();
                },
                isPathRooted: function (path) {
                    throw new System.NotImplementedException();
                },
                getInvalidFileNameChars: function () {
                    throw new System.NotImplementedException();
                },
                getInvalidPathChars: function () {
                    throw new System.NotImplementedException();
                },
                getRandomFileName: function () {
                    throw new System.NotImplementedException();
                },
                validate: function (path) {
                    throw new System.NotImplementedException();
                },
                validate$1: function (path, parameterName) {
                    throw new System.NotImplementedException();
                }
            }
        }
    });
